pkg/ext-proc/backend/provider.go

```diff
@@ -101,31 +104,66 @@ func (p *Provider) Init(refreshPodsInterval, refreshMetricsInterval time.Duratio
 // refreshPodsOnce lists pods and updates keys in the podMetrics map.
 // Note this function doesn't update the PodMetrics value, it's done separately.
 func (p *Provider) refreshPodsOnce() {
-	// merge new pods with cached ones.
-	// add new pod to the map
-	addNewPods := func(k, v any) bool {
-		pod := k.(Pod)
-		if _, ok := p.podMetrics.Load(pod); !ok {
-			new := &PodMetrics{
-				Pod: pod,
-				Metrics: Metrics{
-					ActiveModels: make(map[string]int),
-				},
-			}
-			p.podMetrics.Store(pod, new)
+	pool, err := p.datastore.getInferencePool()
```

Comment:
We have an issue in the current implementation where we only refresh pods every 10s. https://github.com/kubernetes-sigs/gateway-api-inference-extension/blob/528ea6e29973c61df0b0364d6cf7cd60ac324ec8/pkg/ext-proc/main.go#L64

I suggest the following refactoring which should significantly simply the code structure, as well improve performance (gets rid of the pod refresh interval).

- Remove  `refreshPodsOnce` from the provider. 
- Move the `podMetrics` map to the datastore, this should be the single source of truth for all pods the EPP sees.
- The  inferencepool_reconciler.go should trigger pods relisting, and update the `podMetrics` map, if the selector changes.
- There should be a  pod reconciler to update the `podMetrics` map if pods change.

Reply:
My goal is remove refreshPodsOnce completely and just have `refreshMetricsOnce` list directly. I wanted to do this as a quick followup to limit the diff this PR introduces. wdyt?